tips/TIP-0042/tip-0042.md

```diff
@@ -0,0 +1,1230 @@
+</details>
+<table>
+  <tr>
+    <td>
+      <b>Name</b>
+    </td>
+    <td>
+      <b>Type</b>
+    </td>
+    <td>
+      <b>Description</b>
+    </td>
+  </tr>
+  <tr>
+    <td>Address Type</td>
+    <td>uint8</td>
+    <td>Set to <strong>value 32</strong> to denote an <i>Implicit Account Creation Address</i>.</td>
+  </tr>
+  <tr>
+    <td>PubKeyHash</td>
+    <td>ByteArray[32]</td>
+    <td>The raw bytes of the Implicit Account Creation Address which is the BLAKE2b-256 hash of the Ed25519 public key.</td>
+  </tr>
+</table>
+
+An _Implicit Account Creation Address_ is structurally the same as an _Ed25519 Address_, except for the _Address Type_,
+and thus also backed by an Ed25519 key pair.
+
+### Storage Score
+
+- The storage deposit of an _Implicit Account Creation Address_ defines a dedicated Storage Score Offset:
+  `Storage Score Offset Implicit Account Creation Address`. It is designed to match the Storage Score required for a
+  minimal Account Output that contains a Block Issuer Feature when contained in a Basic Output. To calculate this
+  offset, the following Storage Scores are calculated:
+  - Let `Storage Score Offset Implicit Account Creation Address` be
+    `Account Output Score - Basic Output Score + Ed25519 Address Score`, where:
+    - Let `Account Output Score` be the storage score of an Account Output which fulfills the following conditions:
+      - `Address Unlock Condition` must contain an _Ed25519 Address_.
+      - `Features Count == 1`.
+      - A _Block Issuer Feature_ must be present with `Block Issuer Keys Count == 1` and a block issuer key of type
+        _Block Issuer Key Ed25519_.
+      - `Immutable Features Count == 0`.
+    - Let `Basic Output Score` be the storage score of a Basic Output which fulfills the following conditions:
+      - `Unlock Conditions Count == 1`.
+      - The unlock condition is of type `Address Unlock Condition` and contains an `Ed25519 Address`.
+      - `Features Count == 0`.
+    - Let `Ed25519 Address Score` be the storage score of an `Ed25519 Address`.
+      - Note: This is currently `0` but is added here for completeness in case this value changes in the future.
+
+### Creation Syntactic Transaction Validation Rules
+
+To create an implicit account, a Basic Output is created on the output side of a transaction and the `Address` in the
+`Address Unlock Condition` is set to an address of type _Implicit Account Creation Address_.
+
+- If an output in a transaction contains an unlock condition with an _Implicit Account Creation Address_, that output is
+  referred to as an `Implicit Account`, and the transaction is valid only if all of the following conditions hold:
+  - The `Implicit Account` is of type `Basic Output`.
+  - The `Implicit Account`'s `Unlock Conditions Count == 1`.
+  - The unlock condition on the `Implicit Account` is of type `Address Unlock Condition`.
+
+### Conversion Semantic Transaction Validation Rules
```

Comment:
Should it be checked that the address of the ImplicitAccount and the Account are actually the same?

Reply:
I don't think so. In a regular account you can also change the owning Ed25519 Address at any point, and the implicit account -> account transition isn't an exception to that, I think.

Reply:
Fair enough, thanks.